profiling/src/lib.rs

```diff
@@ -692,6 +733,41 @@ extern "C" fn mshutdown(r#type: c_int, module_number: c_int) -> ZendResult {
         profiler.stop();
     }
 
+    #[cfg(feature = "allocation_profiling")]
+    {
+        // At this point ZendMM custom handlers should point to our functions, if they do not we
+        // should not reset the handlers, as this is an indication that:
+        // - some other extension registered custom handlers but did not (yet) unregister itself
+        // - we could not register our custom handlers in minit
+        let mut custom_mm_malloc: Option<zend::VmMmCustomAllocFn> = None;
+        let mut custom_mm_free: Option<zend::VmMmCustomFreeFn> = None;
+        let mut custom_mm_realloc: Option<zend::VmMmCustomReallocFn> = None;
+        unsafe {
+            zend::zend_mm_get_custom_handlers(
+                zend::zend_mm_get_heap(),
+                &mut custom_mm_malloc,
+                &mut custom_mm_free,
+                &mut custom_mm_realloc,
+            );
+        }
+        if custom_mm_free != Some(alloc_profiling_free) ||
+           custom_mm_malloc != Some(alloc_profiling_malloc) ||
+           custom_mm_realloc != Some(alloc_profiling_realloc)
+        {
+            info!("Memory allocation profiling could not be disabled. Please feel free to fill an issue stating the PHP version and installed modules. Most likely another extension registered ZendMM custom handlers but did not unregister properly or your PHP binary was compiled with `ZEND_MM_CUSTOM` being disabled.");
```

Comment:
In this scenario - as noted in an older comment, we should skip unloading our extension by setting our extension handle to `NULL`.
Otherwise, it might be that some other extension forwarding calls to the allocator, like we do, has a pointer to our extension and will just hard crash then.